op-dispute-mon/mon/forecast.go

```diff
@@ -152,6 +158,9 @@ func (f *Forecast) forecastGame(game *monTypes.EnrichedGameData, metrics *foreca
 				"rootClaim", game.RootClaim, "expected", expected)
 		} else {
 			metrics.DisagreeChallengerAhead++
+			if metrics.LatestValidL2Block < game.L2BlockNumber {
+				metrics.LatestValidL2Block = game.L2BlockNumber
+			}
```

Comment:
If we disagree with the root then this isn't a valid proposal and so we probably shouldn't update this metric here.